infra/base-images/base-builder/detect_repo_test.py

```diff
@@ -62,14 +62,50 @@ def test_infer_main_repo(self):
       self.check_commit_with_repo(None, None,
                                   'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', tmp_dir)
 
-  def check_commit_with_repo(self, repo_origin, repo_name, commit, tmp_dir):
+  def test_infer_main_repo_from_name(self):
+    """Tests that the main project repo can be inferred from a repo name."""
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+      # Construct example repos to check for names
+      repo_manager.RepoManager('https://github.com/curl/curl.git', tmp_dir)
+      repo_manager.RepoManager('https://github.com/ntop/nDPI.git', tmp_dir)
+      repo_manager.RepoManager('https://github.com/libarchive/libarchive.git',
+                               tmp_dir)
+      self.check_ref_with_repo('https://github.com/curl/curl.git', 'curl',
+                               tmp_dir)
+      self.check_ref_with_repo('https://github.com/ntop/nDPI.git', 'nDPI',
+                               tmp_dir)
+      self.check_ref_with_repo('https://github.com/libarchive/libarchive.git',
+                               'libarchive', tmp_dir)
+
+  def check_ref_with_repo(self, repo_origin, repo_name, tmp_dir):
     """Checks the detect repo's main method for a specific set of inputs.
 
+      Args:
+        repo_origin: The location of where the git repo url.
+        repo_name: The name of the directory it is cloned to.
+        tmp_dir: The location of the directory of git repos to be searched.
+      """
+    command = [
+        'python3', 'detect_repo.py', '--src_dir', tmp_dir, '--repo_name',
+        repo_name
+    ]
+    out, _ = detect_repo.execute(
+        command, location=os.path.dirname(os.path.realpath(__file__)))
```

Comment:
I think we probably want abspath as well. realpath resolves symbolic links in a way that I don't think we intend here.

Reply:
Done.